CMakeLists.txt

```diff
@@ -222,6 +222,7 @@ else()
   add_compile_flag("-Wswitch") # we explicitly expect this in the code
   if(WIN32)
     add_compile_flag("-D_GNU_SOURCE")
+    add_compile_flag("-D__STDC_FORMAT_MACROS")
```

Comment:
Why this this needed only on win32?

Reply:
I have no idea. I just know that it won't build in MinGW on Windows and it successfully builds on all other platforms. I just took the suggestion from the issue and applied it to only Windows + MinGW.